lib/load-rules.js

```diff
@@ -29,6 +31,11 @@ module.exports = function(rulesDir, cwd) {
         rulesDir = path.resolve(cwd, rulesDir);
     }
 
+    // cache will help performance as IO operation are expensive
+    if (rulesDirCache[rulesDir]) {
```

Comment:
To me, it seems like it's not worth doing this, because we would only be saving a single `fs.readdirSync` call, and this would prevent us from finding additional rules that are added to a `--rulesdir` folder while the same process is running (e.g. in an editor integration). However, I don't feel strongly about it either way.